src/app/components/editors/text-editor/text-editor.component.ts

```diff
@@ -57,25 +68,47 @@ export class TextEditorComponent implements OnInit, OnDestroy {
         });
     }
     public startUpload(data): void {
-        this.disabled = true;
-        const downloadTextMethod = this.uploadDataService.uploadTextData(
-            data.title,
-            data.description,
-        );
+        if (this.titleHeader || this.contentForEditting) {
+            this.disabled = true;
+            const downloadTextMethod = this.uploadDataService.updateTextData(
+                data.title,
+                data.description,
+            );
+
+            const downloadTextSubscription = downloadTextMethod.subscribe(
+                response => {
+                    console.log(response);
+                },
+                error => this.openSnackBar(error),
+                () => {
+                    this.openSnackBar('Text updated');
+                    setTimeout(() => {
+                        this.redirect();
+                        downloadTextSubscription.unsubscribe();
+                    }, 1000);
+                },
+            );
```

Comment:
DRY

Reply:
i remember that we must't reuse same code in different part of App, but i did it for working code instead clear code because of a little time. But i dont forget about this principle.